eclair-core/src/main/scala/fr/acinq/eclair/db/jdbc/JdbcUtils.scala

```diff
@@ -60,6 +60,42 @@ trait JdbcUtils {
     }
   }
 
+  private def createVersionTable(statement: Statement): Unit = {
+    statement.executeUpdate("CREATE TABLE IF NOT EXISTS versions (db_name TEXT NOT NULL PRIMARY KEY, version INTEGER NOT NULL)")
+  }
+
+  /**
+   * Several logical databases (channels, network, peers) may be stored in the same physical sqlite database.
+   * We keep track of their respective version using a dedicated table. The version entry will be created if
+   * there is none but will never be updated here (use setVersion to do that).
+   */
+  def getVersion(statement: Statement, db_name: String): Option[Int] = {
+    createVersionTable(statement)
+    // if there was a previous version installed, this will return a different value from current version
+    val rs = statement.executeQuery(s"SELECT version FROM versions WHERE db_name='$db_name'")
+    if (rs.next()) Some(rs.getInt("version")) else None
+  }
+
+  /**
+   * Updates the version for a particular logical database, it will overwrite the previous version.
+   *
+   * NB: we could define this method in [[fr.acinq.eclair.db.sqlite.SqliteUtils]] and [[fr.acinq.eclair.db.pg.PgUtils]]
+   *     but it would make testing more complicated because we need to use one or the other depending on the backend.
+   */
+  def setVersion(statement: Statement, db_name: String, newVersion: Int): Unit = {
+    createVersionTable(statement)
+    statement.getConnection match {
+      case _: SQLiteConnection =>
+        // if there was no version for the current db, then insert the current version
+        statement.executeUpdate(s"INSERT OR IGNORE INTO versions VALUES ('$db_name', $newVersion)")
+        // if there was an existing version, then previous step was a no-op, now we overwrite the existing version
+        statement.executeUpdate(s"UPDATE versions SET version=$newVersion WHERE db_name='$db_name'")
+      case _ => // if it isn't an sqlite connection, we assume it's postgres
+        // insert or update the version
+        statement.executeUpdate(s"INSERT INTO versions VALUES ('$db_name', $newVersion) ON CONFLICT (db_name) DO UPDATE SET version = EXCLUDED.version ;")
```

Comment:
Why did you change the SQL here? It was `s"UPDATE versions SET version=$newVersion WHERE db_name='$db_name'"`, I find the old version more readable.

Reply:
This replaces two statements:
```scala
    statement.executeUpdate(s"INSERT INTO versions VALUES ('$db_name', $currentVersion) ON CONFLICT DO NOTHING")
...
    statement.executeUpdate(s"UPDATE versions SET version=$newVersion WHERE db_name='$db_name'")
```

This is the recommended pattern according to postgres doc
(https://www.postgresql.org/docs/current/plpgsql-control-structures.html#PLPGSQL-UPSERT-EXAMPLE):

> It is recommended that applications use INSERT with ON CONFLICT DO
UPDATE rather than actually using this pattern.

Reply:
Understood, previously `setVersion` assumed that the version was already there (but there still was a useless `CREATE TABLE IF NOT EXISTS`) and needed updating but now it may not be there in the first place and we need `INSERT`.